tests/scenario/test_grafana_cloud_integrator_lib.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from charms.grafana_cloud_integrator.v0.cloud_config_requirer import (
+    CloudConfigAvailableEvent,
+    CloudConfigRevokedEvent,
+    GrafanaCloudConfigRequirer,
+)
+from ops import CharmBase, Framework
+from scenario import Context, Relation, State
+
+
+class MyCharm(CharmBase):
+    def __init__(self, framework: Framework):
+        super().__init__(framework)
+        self.cloud = GrafanaCloudConfigRequirer(self)
+
+
+@pytest.fixture()
+def context():
+    return Context(
+        charm_type=MyCharm,
+        meta={
+            "name": "my-charm",
+            "requires": {
+                "grafana-cloud-config": {"interface": "grafana_cloud_config", "limit": 1}
+            },
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "is_leader",
+    [
+        (True,),
+        (False,),
+    ],
+)
+def test_requirer_emits_cloud_config_available_event_on_relation_changed(is_leader, context):
+    # GIVEN a grafana-cloud-config relation
+    grafana_cloud_config_relation = Relation("grafana-cloud-config")
+    # AND GIVEN leadership/non-leadership
+    state = State(leader=is_leader, relations=[grafana_cloud_config_relation])
+
+    # WHEN the grafana-cloud-config relation changes
+    context.run(grafana_cloud_config_relation.changed_event, state)
+
+    # THEN the CloudConfigAvailableEvent event is emitted
+    any(event for event in context.emitted_events if isinstance(event, CloudConfigAvailableEvent))
+
+
+@pytest.mark.parametrize(
+    "is_leader",
+    [
+        (True,),
+        (False,),
+    ],
+)
+def test_requirer_emits_cloud_config_revoked_event_on_relation_broken(is_leader, context):
+    # GIVEN a grafana-cloud-config relation
+    grafana_cloud_config_relation = Relation("grafana-cloud-config")
+    # AND GIVEN leadership/non-leadership
+    state = State(leader=is_leader, relations=[grafana_cloud_config_relation])
+
+    # WHEN the grafana-cloud-config relation changes
+    context.run(grafana_cloud_config_relation.broken_event, state)
```

Comment:
Per [`DAMP`](https://discourse.charmhub.io/t/best-practices-for-testing-charms/15172#a-test-should-be-obvious-upon-inspection-damp-7), how about:
```suggestion
    MyCharm.get_context().run(grafana_cloud_config_relation.broken_event, state)
```

Reply:
I edited the name of the fixture to be more descriptive.  Agreed that `context` alone is too broad, but fixtures still seem appropriate (see [this comment](https://github.com/canonical/grafana-cloud-integrator/pull/13#discussion_r1720049863))

Reply:
Resolving given @Abuelodelanada's review, but if anyone disagrees we can open another issue and fix it